### PR TITLE
Updated api_swagger.json as per version 1.1 of the FSPIOP spec and re…

### PR DESCRIPTION
**Pull request by ndonnan**
…moved unused type definitions

### CODE REVIEW

**Comment by elnyry** [comment]
@ndonnan - thanks for taking this up.

Regarding the Swagger, I think so far, the strategy has been to use the same copy of the API across services, for ease of maintenance, even though some data items may not be useful in a particular service (such as transfers resources not being used in ALS). what do you think?

**Comment by ndonnan** [comment]
> @ndonnan - thanks for taking this up.
> 
> Regarding the Swagger, I think so far, the strategy has been to use the same copy of the API across services, for ease of maintenance, even though some data items may not be useful in a particular service (such as transfers resources not being used in ALS). what do you think?
> 
> Also, not all changes are still included I think for v1.1, not sure if you're still working on it.

@elnyry Ok, I understand, I probably wouldn't duplicate types that aren't used in a particular service since I would think it would add more maintenance e.g. if the transfer definition changed you would need to update it in the ALS even though there are no services in the ALS that use it, or if it wasn't updated someone in the future may use it without realising it hasn't been kept up to date. Also if you use tools such as the Swagger generator to generate a java client for example you will end up with a whole bunch of classes that aren't used, confusing users as to why they are part of the client.

I have worked through all the parties and participants endpoints and can't find anything missing, the optional extensionList was added to PartyIdInfo and ParticipantsTypeIDSubIDPostRequest types which are reused by the services. Let me know if there is anything you have picked up.

**Comment by elnyry** [comment]
Thanks @ndonnan - if PartyIdInfo and ParticipantsTypeIDSubIDPostRequest are updated, I think we're good.

File: src/interface/api_swagger.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "swagger": "2.0",
   "info": {
-    "version": "1.0.1",
+    "version": "1.1",
     "title": "Open API for FSP Interoperability (FSPIOP) (Implementation Friendly Version)",
-    "description": "Based on API Definition.docx updated on 2018-03-13 Version 1.0. Note - The API supports a maximum size of 65536 bytes (64 Kilobytes) in the HTTP header.",
+    "description": "Based on API Definition.docx Version 1.1. Note - The API supports a maximum size of 65536 bytes (64 Kilobytes) in the HTTP header.",
```
Comment (elnyry) [inline src/interface/api_swagger.json:6]:
```suggestion
    "description": "Based on API Definition.docx Version 1.1 that was confirmed on 2020-05-19. Note - The API supports a maximum size of 65536 bytes (64 Kilobytes) in the HTTP header.",
```

File: src/interface/api_swagger.json [hunk 2]
```diff
@@ -2438,21 +2345,6 @@
       "maxLength": 128,
       "description": "Extension value."
     },
-    "FirstName": {
```
Comment (elnyry) [inline src/interface/api_swagger.json:2441]: I'm looking only at the changes here, but the FirstName, LastName, Name, MiddleName, FspId are things we need in /participants and /parties calls, isn't it?

File: src/interface/api_swagger.json [hunk 3]
```diff
@@ -2399,18 +2318,6 @@
       "pattern": "^(?:[1-9]\\d{3}-(?:(?:0[1-9]|1[0-2])-(?:0[1-9]|1\\d|2[0-8])|(?:0[13-9]|1[0-2])-(?:29|30)|(?:0[13578]|1[02])-31)|(?:[1-9]\\d(?:0[48]|[2468][048]|[13579][26])|(?:[2468][048]|[13579][26])00)-02-29)$",
       "description": "The API data type Date is a JSON String in a lexical format that is restricted by a regular expression for interoperability reasons. This format, as specified in ISO 8601, contains a date only. A more readable version of the format is yyyy-MM-dd. Examples - \"1982-05-23\", \"1987-08-05”"
     },
-    "DateOfBirth": {
```
Comment (elnyry) [inline src/interface/api_swagger.json:2402]: I'm looking only at the changes here, but isn't this something we need in /participants and /parties calls?

File: src/interface/api_swagger.json [hunk 4]
```diff
@@ -2565,317 +2369,6 @@
...
     },
-    "PartyName": {
-      "title": "PartyName",
-      "type": "string",
-      "minLength": 1,
-      "maxLength": 128,
-      "description": "Name of the Party. Could be a real name or a nickname."
-    },
-    "PartySubIdOrType": {
```
Comment (elnyry) [inline src/interface/api_swagger.json:2575]: Suspect we need this as well

File: src/interface/api_swagger.json [hunk 5]
```diff
@@ -2565,317 +2369,6 @@
...
     },
-    "PartyName": {
```
Comment (elnyry) [inline src/interface/api_swagger.json:2568]: Suspect we need this as well

File: src/interface/api_swagger.json [hunk 6]
```diff
@@ -2565,317 +2369,6 @@
...
-      "description": "Name of the Party. Could be a real name or a nickname."
-    },
-    "PartySubIdOrType": {
-      "title": "PartySubIdOrType",
-      "type": "string",
-      "minLength": 1,
-      "maxLength": 128,
-      "description": "Either a sub-identifier of a PartyIdentifier, or a sub-type of the PartyIdType, normally a PersonalIdentifierType."
-    },
-    "PersonalIdentifierType": {
```
Comment (elnyry) [inline src/interface/api_swagger.json:2582]: Suspect we need this as well

File: src/interface/api_swagger.json (hunk 5 above)
Comment (ndonnan) [reply src/interface/api_swagger.json:2568]: Yip, I just went a removed all unused type definitions, the FirstName, DateOfBirth etc weren't being used. I will go back through the Swagger definition and include them where needed.

File: src/interface/api_swagger.json (hunk 2 above)
Comment (ndonnan) [reply src/interface/api_swagger.json:2441]: I have referenced these correctly

File: src/interface/api_swagger.json (hunk 3 above)
Comment (ndonnan) [reply src/interface/api_swagger.json]: I have referenced these correctly

File: src/interface/api_swagger.json (hunk 4 above)
Comment (ndonnan) [reply src/interface/api_swagger.json:2575]: I have referenced these correctly

File: src/interface/api_swagger.json (hunk 6 above)
Comment (ndonnan) [reply src/interface/api_swagger.json:2582]: I have referenced these correctly

**Pull request merged**